### PR TITLE
Port hybrid angle, dihedral, improper topology styles to Kokkos

### PR DESCRIPTION
**Pull request by megmcca**
**Summary**
This PR adds Kokkos support for hybrid angle/dihedral/improper styles similar to PR #4167.  

**Author(s)**

Meg McCarthy, Stan Moore (Sandia National Labs)

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

Yes

**Implementation Notes**

Tested on NVIDIA RTX A4000

### CODE REVIEW

**Comment by akohlmey** [comment]
@megmcca FYI, I filled out the pull request flags on the right side panel (assignee, Project, labels) for you.
Our protocol is the following:
- while you are working on a pull request, you assign it to yourself. That indicates to others to not make any changes.
- when it is ready for review, you unassign yourself and possibly assign it to the "primary" reviewer (or wait for the "GitHub caretaker" to do it). In this case it is obviously @stanmoore1
- if desired or meaningful, you can then also assign reviewers in addition to the auto-assigned ones
- when it is approved, it should be assigned to the person doing the merging, i.e. currently me. I may want to make some cosmetic changes or confirm that the documentation is updated accordingly etc. before finally merging.

File: src/KOKKOS/angle_hybrid_kokkos.cpp [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    k_nanglelist.modify_device();
+    k_nanglelist.sync_host();
+
+    maxangle_all = 0;
+    for (int m = 0; m < nstyles; m++)
+      if (h_nanglelist[m] > maxangle_all)
+        maxangle_all = h_nanglelist[m] + EXTRA;
+
+    if (k_anglelist.d_view.extent(1) < maxangle_all)
+      MemKK::realloc_kokkos(k_anglelist, "angle_hybrid:anglelist", nstyles, maxangle_all, 3);
```
Comment (stanmoore1) [inline src/KOKKOS/angle_hybrid_kokkos.cpp]: @megmcca also need a corresponding increase to the size of the view here.

Comment (megmcca) [reply src/KOKKOS/angle_hybrid_kokkos.cpp]: got it! tested it and output energies for `angle hybrid harmonic charmm` on non-Kokkos, cpu-Kokkos, and gpu-Kokkos compiles match each other.

File: src/KOKKOS/bond_hybrid_kokkos.cpp [hunk 2]
```diff
@@ -153,11 +153,11 @@ void BondHybridKokkos::compute(int eflag, int vflag)
...
   allocated = 1;
-  int n = atom->nbondtypes;
+  int np1 = atom->nbondtypes + 1;
 
-  memoryKK->create_kokkos(k_map, map, n + 1, "bond:map");
-  memory->create(setflag, n + 1, "bond:setflag");
-  for (int i = 1; i <= n; i++) setflag[i] = 0;
+  memoryKK->create_kokkos(k_map, map, np1, "bond:map");
+  memory->create(setflag, np1, "bond:setflag");
+  for (int i = 1; i <= np1; i++) setflag[i] = 0;
```
Comment (stanmoore1) [inline src/KOKKOS/bond_hybrid_kokkos.cpp]: @megmcca I think this loop goes one too many iterations, should be `i < np1` or `i <= n` as in the host code, same for other hybrid styles.

**Review by stanmoore1 — APPROVED** [review]

**Review by sjplimp — APPROVED** [review]

**Review by akohlmey — CHANGES_REQUESTED** [review]
@megmcca Please do the following:
- merge with the changes from upstream
- go to the src folder, type make fix-whitespace and commit the resulting changes
- update the documentation:
   + add "(k)" to the hybrid styles in doc/src/Commands_bond.rst
   + add index entries and the include for accelerated style to the documentation for angle/dihedral/improper style hybrid. See doc/src/bond_hybrid.rst for comparison. When you do "make html" in the doc folder (or "make check" in the src folder) you trigger a check for the missing pieces.

**Review by akohlmey — APPROVED** [review]

**Pull request merged**